src/rest/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+#
+#  Copyright (c) 2020, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+add_library( otbr-rest 
+    rest_web_server.cpp
+    connection.cpp
+    resource.cpp
+    json.cpp
+    parser.cpp
+    request.cpp
+    response.cpp
+)
+
+target_include_directories( otbr-rest 
+    PUBLIC
+        ${PROJECT_SOURCE_DIR}/third_party/http-parser/repo
+    PRIVATE
+        ${PROJECT_SOURCE_DIR}/third_party/cJSON/repo        
+)
```

Comment:
For http-parser, it works fine. For cJSON, its native CMakeList file uses install instead of target_include_directories to make its header file could be found. but seems the install command does not work at my side, I will move on  and address this later.

Reply:
You can create a CMakeLists.txt file under exactly `third_party/cJSON`, with below content:
```cmake
add_subdirectory(repo)

target_include_directories(cjson
    PUBLIC
        ${CMAKE_CURRENT_SOURCE_DIR}/repo
)
```

And add `third_party/cjson`,  instead of `third_paerty/cjson/repo`,  as subdirectory in `third_party/CMakeLists.txt`.